### ISSUE TITLE
Crash when analyzing a custom element with a field that's a mixin

### ISSUE THREAD

**Issue by rictic** [runem/web-component-analyzer#60]
Analyzing this code:

```typescript
type Constructor<T> = {
  new (...args: any[]): T
};
type ConstructorOne = Constructor<{someMethod(): void;}>;
type ConstructorTwo = Constructor<{someMethod(): void;}>;

type Combined = ConstructorOne&ConstructorTwo;
declare const Combined: Combined;

declare class ExtendsCombined extends Combined {}

export class C extends HTMLElement {
  field!: ExtendsCombined;
}
customElements.define('s-e', C);
```

results in the following crash:

```
TypeError: Cannot read property 'kind' of undefined
    at Object.isBindingElement (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/typescript/lib/typescript.js:13739:21)
    at getCombinedFlags (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/typescript/lib/typescript.js:13081:16)
    at Object.getCombinedModifierFlags (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/typescript/lib/typescript.js:13098:16)
    at getModifiersFromDeclaration (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/ts-simple-type/lib/index.cjs.js:230:37)
    at /Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/ts-simple-type/lib/index.cjs.js:499:28
    at Array.map (<anonymous>)
    at toSimpleTypeInternal (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/ts-simple-type/lib/index.cjs.js:497:63)
    at toSimpleTypeInternalCaching (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/ts-simple-type/lib/index.cjs.js:291:33)
    at toSimpleType (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/ts-simple-type/lib/index.cjs.js:257:12)
    at Object.isAssignableToSimpleTypeKind (/Users/rictic/open/fuck/repro-mixin-simple-type/node_modules/ts-simple-type/lib/index.cjs.js:654:45)
```

This is a minimal reduction of a bug from a LitElement class that had a field that was an array of PolymerElements, whose type is defined by application of a number of [class mixins](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-mix-in-classes).

**Comment by rictic** [comment]
The issue seems to be that `typescript.Symbol#valueDeclaration` can be undefined (despite its type not including undefined). The symbol for `someMethod` doesn't have a valueDeclaration, perhaps because of some weirdness in the way that TypeScript handles combining constructors.

The simple fix is to filter out property symbols that have an undefined valueDeclaration in the class section of `toSimpleTypeInternal`, though that will result in dropping some fields from classes. Fortunately, these are not fields that are likely to be used much from templates, so that may be fine.

**Comment by rictic** [comment]
This should probably also be repro'd against the TypeScript compiler API and filed against them as well.

**Comment by runem** [comment]
Thanks for creating the issue and PR! I have now published the fix with ts-simple-type version 0.3.5 :tada:
